pyomo/contrib/doe/doe.py

```diff
@@ -297,39 +272,27 @@ def _optimize_stochastic_program(self, m):
         analysis_optimize = FisherResults(list(self.param.keys()), self.measure, jacobian_info=None, all_jacobian_info=jac_optimize,
                                         prior_FIM=self.prior_FIM)
         # for simultaneous mode, FIM and Jacobian are extracted with extract_FIM()
-        analysis_optimize.calculate_FIM(self.design_timeset, result=result_doe)
+        analysis_optimize.calculate_FIM(self.design_name, result=result_doe)
         analysis_optimize.model = m
 
-        # record optimization time
-        analysis_optimize.solve_time = time_solve2
-
         return analysis_optimize
 
 
 
-    def compute_FIM(self, design_values, mode='sequential_finite', FIM_store_name=None, specified_prior=None,
+    def compute_FIM(self, mode=calculation_mode.direct_kaug, FIM_store_name=None, specified_prior=None,
                     tee_opt=True, scale_nominal_param_value=False, scale_constant_value=1,
                     store_output = None, read_output=None, extract_single_model=None,
-                    formula='central', step=0.001,
-                    objective_option='det'):
+                    formula=finite_difference_step.central, step=0.001):
         """
         This function solves a square Pyomo model with fixed design variables to compute the FIM.
-        It calculates FIM with sensitivity information from four modes:
-            1.  sequential_finite: Calculates a one scenario model multiple times for multiple scenarios. 
-            Sensitivity info estimated by finite difference
-            2.  sequential_sipopt: calculate sensitivity by sIPOPT [Experimental]
-            3.  sequential_kaug: calculate sensitivity by k_aug [Experimental]
-            4.  direct_kaug: calculate sensitivity by k_aug with direct sensitivity
-
-        "Simultaneous_finite" mode is not included in this function.
+        It calculates FIM with sensitivity information from the Enum calculation_mode with two modes:
```

Comment:
This function calculates the Fisher information matrix (FIM) using sensitivity information obtained from two possible modes (defined by the calculation_mode Enum):